--- origsrc/libmusicbrainz-5.0.1/CMakeLists.txt	2012-05-16 14:48:34.000000000 -0500
+++ src/libmusicbrainz-5.0.1/CMakeLists.txt	2012-09-05 23:59:58.630801800 -0500
@@ -32,7 +32,7 @@ CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DI
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_SOURCE_DIR}/config.h)
 
 FILE(GLOB headers ${CMAKE_CURRENT_SOURCE_DIR}/include/musicbrainz5/*.h)
-INSTALL(FILES ${headers} include/musicbrainz5/mb5_c.h DESTINATION ${INCLUDE_INSTALL_DIR}/musicbrainz5)
+INSTALL(FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/include/musicbrainz5/mb5_c.h DESTINATION ${INCLUDE_INSTALL_DIR}/musicbrainz5)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libmusicbrainz5.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 
 ADD_SUBDIRECTORY(src)
--- origsrc/libmusicbrainz-5.0.1/examples/CMakeLists.txt	2012-05-16 14:48:34.000000000 -0500
+++ src/libmusicbrainz-5.0.1/examples/CMakeLists.txt	2012-09-05 23:58:04.627648500 -0500
@@ -1,4 +1,4 @@
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../include)
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR}/../include ${CMAKE_CURRENT_SOURCE_DIR}/../include)
 LINK_LIBRARIES(musicbrainz5 ${NEON_LIBRARIES})
 
 ADD_EXECUTABLE(cdlookup cdlookup.cc)
--- origsrc/libmusicbrainz-5.0.1/src/CMakeLists.txt	2012-09-05 23:31:59.772600800 -0500
+++ src/libmusicbrainz-5.0.1/src/CMakeLists.txt	2012-09-05 23:54:26.644459500 -0500
@@ -1,5 +1,5 @@
 SET(CMAKE_INCLUDE_CURRENT_DIR ON)
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/.. ${CMAKE_CURRENT_SOURCE_DIR}/../include ${NEON_INCLUDE_DIR})
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/.. ${CMAKE_CURRENT_BINARY_DIR}/../include ${CMAKE_CURRENT_SOURCE_DIR}/../include ${NEON_INCLUDE_DIR})
 
 SET(_sources Alias.cc Annotation.cc Artist.cc ArtistCredit.cc Attribute.cc CDStub.cc Collection.cc
 	Disc.cc Entity.cc FreeDBDisc.cc HTTPFetch.cc ISRC.cc Label.cc LabelInfo.cc Lifespan.cc List.cc
@@ -12,6 +12,7 @@ ADD_EXECUTABLE(make-c-interface make-c-i
 
 ADD_CUSTOM_COMMAND(
 	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/mb5_c.cc ${CMAKE_CURRENT_BINARY_DIR}/mb5_c.h ${CMAKE_CURRENT_BINARY_DIR}/../include/musicbrainz5/mb5_c.h
+	COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/../include/musicbrainz5
 	COMMAND make-c-interface ${CMAKE_CURRENT_SOURCE_DIR} cinterface.xml ${CMAKE_CURRENT_BINARY_DIR} mb5_c.cc mb5_c.h
 	COMMAND ${CMAKE_COMMAND} -E copy_if_different 	${CMAKE_CURRENT_BINARY_DIR}/mb5_c.h ${CMAKE_CURRENT_BINARY_DIR}/../include/musicbrainz5
 	DEPENDS make-c-interface cinterface.xml *.inc
--- origsrc/libmusicbrainz-5.0.1/src/make-c-interface.cc	2012-05-16 14:48:34.000000000 -0500
+++ src/libmusicbrainz-5.0.1/src/make-c-interface.cc	2012-09-05 23:49:27.424046600 -0500
@@ -40,6 +40,8 @@ void ProcessClass(const XMLNode& Node, s
 void ProcessList(const XMLNode& Node, std::ofstream& Source, std::ofstream& Include);
 void ProcessDeclare(const XMLNode& Node, std::ofstream& Source, std::ofstream& Include);
 
+std::string SrcDir;
+
 int main(int argc, const char *argv[])
 {
 	for (int count=0;count<argc;count++)
@@ -49,6 +51,7 @@ int main(int argc, const char *argv[])
 
 	if (argc==6)
 	{
+		SrcDir = std::string(argv[1]);
 		std::string XMLFile=std::string(argv[1])+"/"+argv[2];
 		XMLResults Results;
 		XMLNode TopNode=XMLNode::parseFile(XMLFile.c_str(),"cinterface",&Results);
@@ -162,7 +165,7 @@ void ProcessBoilerplate(const XMLNode& N
 
 	if (Node.isAttributeSet("file"))
 	{
-		std::string FileName=Node.getAttribute("file");
+		std::string FileName=SrcDir + "/" + Node.getAttribute("file");
 		std::ifstream InFile(FileName.c_str());
 		if (InFile.is_open())
 			*File << InFile.rdbuf() << std::endl;
--- origsrc/libmusicbrainz-5.0.1/tests/CMakeLists.txt	2012-05-16 14:48:34.000000000 -0500
+++ src/libmusicbrainz-5.0.1/tests/CMakeLists.txt	2012-09-05 23:57:33.817606000 -0500
@@ -1,4 +1,4 @@
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../include)
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR}/../include ${CMAKE_CURRENT_SOURCE_DIR}/../include)
 LINK_LIBRARIES(musicbrainz5 ${NEON_LIBRARIES})
 ADD_EXECUTABLE(mbtest mbtest.cc)
 ADD_EXECUTABLE(ctest ctest.c)
